tests/core/test_accounts.py

```diff
@@ -322,6 +322,44 @@ def test_eth_account_sign(acct, message, key, expected_bytes, expected_hash, v,
     assert account.signHash(msghash) == signed
 
 
+@pytest.mark.parametrize(
+    'message, message_type',
+    (
+        ("hello world", "text"),
+        (b'hello world', "primitive"),
+        ("68656c6c6f20776f726c64", "hexstr"),
+    )
+)
+def test_eth_account_sign_data_with_intended_validator(message, message_type, acct):
+    account = acct.create()
+
+    kwargs = {message_type: message}
+
+    with pytest.raises(TypeError):
+        # Raise TypeError if the address is more than 20 bytes
+        defunct_hash_message(
+            **kwargs,
+            signature_version=b'\x00',
+            version_specific_data=account.address + "12345",
+        )
+    with pytest.raises(TypeError):
+        # Raise TypeError if the address is less than 20 bytes
```

Comment:
We like highly granular tests.  The reasoning being that it's better to not have multiple failures hiding behind a single failure.  I.E. Were both of these assertions to be failing (the method accepted both short and too-long addresses, then you wouldn't see the short-address failure until you fixed the long-address failure.

Ideally we want individual test cases for each of the *conditions* we test which suggest this test function should be split into three:

- test too-short address
- test too-long address
- test *valid* address.